_posts/2016-08-26-babili.md

```diff
@@ -0,0 +1,183 @@
+---
+layout: post
+title:  "Releasing Babili (beta)"
+author: Henry Zhu
+date:   2016-08-26 09:30:00
+categories: announcements
+share_text: "Releasing Babili (beta)"
+---
+
+# Babili (babel-minify)
+
+We released [babili](https://github.com/babel/babili) as beta (0.0.1) yesterday under an MIT license!
+
+There are a lot of (valid) questions about why a new minifier is a good idea, so this post should help with that. We also have a [#minify](https://babeljs.slack.com/messages/minify/) slack room!
+
+TL;DR: Babili can output to ES2015+ code so you don't need to transpile and then minify if it's not necessary. Babili is also modular/flexible (it's a babel preset and thus supports user plugins) and can be used as a preset or cli tool. Babili should also be able to do ES2015+ specific optimizations.
+
+## Prounciation
+
+```bash
+# sounds like "bah billy"
+say babili
+```
+
+
+> If you can't remember the name, babel-minify works too.
+
+## Why minify?
+
+[Minification](https://en.wikipedia.org/wiki/Minification_(programming)) removes unncessary characeters without changing its functionality. The basics of minification remove things like comments, whitespace, newlines, and extra parentheses. You can also simplify code, transform code to smaller equivalents, remove unused code, etc.
+
+Minification is useful because it reduces the amount of code transferred from the server to the client. Because users download less, the page can load faster.
+
+## Current minifiers
+
+Tools such as [uglify](https://github.com/mishoo/UglifyJS2) don't currently support targeting the latest version of Javascript ([yet](https://github.com/mishoo/UglifyJS2/issues/448)).
+
+We currently to use tools like Babel to compile ES2015 code down to ES5 code to support older browsers. Then we use something like uglify to cut down on the bundle size.
+
+As browsers implement more ES2015 features and we drop support for older browser versions, there is a sliding window of the version of javascript you write in and the target javascript version you compile/minify to. However because uglify cannot parse ES2015, you would still have to compile down to ES5 anyway.
+
+## Babili
+
+That's where Babili comes in.
+
+Babili is ES2015+ aware because it is built using the Babel toolchain. More specifically, it is a set of babel plugins + a preset just like with the `es2015` preset.
+
+Like described in [Not Born to Die](http://babeljs.io/blog/2015/02/15/not-born-to-die) when Babel was renamed from 6to5, Babel was originally a specific transpiler for ES2016 to ES5. A transpiler is just a compiler so the same concepts to transform arrow functions can work to minify code.
```

Comment:
maybe it should just be removed..?

`Babel was originally a specific transpiler for ES2016 to ES5 (called 6to5). Babili is an example of the many kinds of projects that Babel that help with.`